src/main/java/se/liu/ida/hefquin/engine/wrappers/graphqlwrapper/impl/SPARQL2GraphQLTranslatorImpl.java

```diff
@@ -0,0 +1,505 @@
+                    return sgpType;
+                }
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * @return a map consisting of what can be used as parameters from the given @param sgp
+     * The predicate from a TP needs to be a property URI and the object needs to be a literal
+     */
+    protected final Map<String,LiteralLabel> getSgpParameters(final Set<TP> sgp, final RdfViewConfiguration config){
+        Map<String,LiteralLabel> parameters = new HashMap<>();
+        for(final TP t : sgp){
+            final Node predicate = t.getTp().asJenaTriple().getPredicate();
+            final Node object = t.getTp().asJenaTriple().getObject();
+
+            if(predicate.isURI() && predicate.getURI().startsWith(config.getPropertyPrefix())){
+                if(object.isLiteral() && object.getLiteral().isWellFormed()){
+                    // Remove prefix and suffix before adding to map
+                    final String s = removePrefix(predicate.toString(),config);
+                    final String parameterName = removeSuffix(s);
+                    parameters.put(parameterName,object.getLiteral());
+                }
+            }
+        }
+        return parameters;
+    }
+
+
+    protected boolean hasNecessaryParameters(final Set<String> sgpParameters, final Set<String> entrypointParameters, 
+            final boolean checkAll){
+        
+        if(sgpParameters.isEmpty()){
+            return false;
+        }
+
+        if(checkAll){
+            return sgpParameters.containsAll(entrypointParameters);
+        }
+        else{
+            for(final String paramName : sgpParameters){
+                if(entrypointParameters.contains(paramName)){
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Utility function used to convert a literal value to a valid jsonvalue
+     */
+    protected JsonValue literalToJsonValue(LiteralLabel literal){
+        final Class<?> type = literal.getValue().getClass();
+
+        if(type.equals(int.class)){
+            return JsonNumber.value((int) literal.getValue());
+        }
+        else if(type.equals(boolean.class)){
+            return new JsonBoolean((boolean) literal.getValue());
+        }
+        else if(type.equals(double.class) || type.equals(float.class)){
+            return JsonNumber.value((double) literal.getValue());
+        }
+        else{
+            return new JsonString((String) literal.getValue());
+        }
+    }
+
+    /**
+     * Wrapper class for a TriplePattern that includes an integer id
+     */
+    private class TP {
+        private final int id;
+        private final TriplePattern tp;
+
+        public TP(final int id, final TriplePattern tp) {
+            this.id = id;
+            this.tp = tp;
+        }
+
+        public final int getId() {
+            return id;
+        }
+
+        public final TriplePattern getTp(){
```

Comment:
You are returning a `TriplePattern` object, not a `TP` object, so name the method accordingly.
```suggestion
        public final TriplePattern getTriplePattern(){
```